packages/application-generic/src/logging/index.ts

```diff
@@ -88,7 +83,6 @@ export function createNestLoggingModuleOptions(settings: ILoggerSettings) {
   return {
     pinoHttp: {
```

Comment:
@Cliftonz probably lost in my previous comments, but why are we implementing masking from scratch and not using the built in redact for pino? https://github.com/pinojs/pino/blob/master/docs/redaction.md

Reply:
Because the redact functionality requires that you know the exact path to react in every json object and it just blanket redacts without any knowledge of what we are masking. For example, email should be masked differently then a password.

Reply:
I attempted to implement this but the wildcard pathing was not working. @scopsy If you can take a look at this I think that would be best.
In this pr, I have kept my original implementation and made sure that the request logging processing is only done if log level is set to debug and uses promises to ensure non-synchronous blocking.